### ISSUE TITLE
Can't import mysql-connector==2.1.4 in pex environment

### ISSUE THREAD

**Issue by quakelee** [pantsbuild/pex#351]
Hi guys,
My app depends on mysql-connector, so I installed it in my virtualenv and I can use it without problem when I run it directly with python

my python environment like this
python
Python 2.7.5 (default, Nov  6 2016, 00:28:07)
[GCC 4.8.5 20150623 (Red Hat 4.8.5-11)] on linux2
Type "help", "copyright", "credits" or "license" for more information.
>>> import mysql.connector
>>> import pip
>>> installed_packages = pip.get_installed_distributions()
>>> installed_packages_list = sorted(["%s==%s" % (i.key, i.version)
...      for i in installed_packages])
>>> print(installed_packages_list)
['backports.ssl-match-hostname==3.4.0.2', 'cffi==1.9.1', 'chardet==2.2.1', 'click==6.7', 'colorama==0.3.7', 'cryptography==1.7.1', 'docopt==0.6.2', 'enum34==1.1.6', 'fabric==1.13.1', 'idna==2.2', 'iniparse==0.4', 'ipaddress==1.0.18', 'jinja2==2.9.4', 'kaptan==0.5.8', 'kitchen==1.1.1', 'libtmux==0.6.1', 'markupsafe==0.23', 'mysql-connector-python-rf==2.2.1', 'mysql-connector==2.1.4', 'mysql-python==1.2.5', 'paramiko==2.1.1', 'pex==1.1.20', 'pip==9.0.1', 'prettytable==0.7.2', 'psutil==5.0.1', 'pyasn1==0.1.9', 'pycparser==2.17', 'pycurl==7.19.0', 'pygpgme==0.3', 'pyliblzma==0.5.3', 'pyxattr==0.5.1', 'pyyaml==3.12', 'records==0.5.0', 'requests==2.12.5', 'setuptools==30.4.0', 'six==1.10.0', 'sqlalchemy==1.1.5', 'tablib==0.11.3', 'tmuxp==1.2.4', 'wheel==0.24.0']

but I can't import mysql.connector in pex environment, I tried a lot of way, and google many times, still have no clue about this

here is my trying

pip freeze
backports.ssl-match-hostname==3.4.0.2
cffi==1.9.1
chardet==2.2.1
click==6.7
colorama==0.3.7
cryptography==1.7.1
docopt==0.6.2
enum34==1.1.6
Fabric==1.13.1
idna==2.2
iniparse==0.4
ipaddress==1.0.18
Jinja2==2.9.4
kaptan==0.5.8
kitchen==1.1.1
libtmux==0.6.1
MarkupSafe==0.23
mysql-connector==2.1.4
mysql-connector-python-rf==2.2.1
MySQL-python==1.2.5
paramiko==2.1.1
pex==1.1.20
prettytable==0.7.2
psutil==5.0.1
pyasn1==0.1.9
pycparser==2.17
pycurl==7.19.0
pygpgme==0.3
pyliblzma==0.5.3
pyxattr==0.5.1
PyYAML==3.12
records==0.5.0
requests==2.12.5
six==1.10.0
SQLAlchemy==1.1.5
tablib==0.11.3
tmuxp==1.2.4

# pex -r <(pip freeze) pip
Python 2.7.5 (default, Nov  6 2016, 00:28:07)
[GCC 4.8.5 20150623 (Red Hat 4.8.5-11)] on linux2
Type "help", "copyright", "credits" or "license" for more information.
(InteractiveConsole)
>>> import pip
>>> installed_packages = pip.get_installed_distributions()
>>> installed_packages_list = sorted(["%s==%s" % (i.key, i.version)
...      for i in installed_packages])
>>> print(installed_packages_list)
['backports.ssl-match-hostname==3.4.0.2', 'cffi==1.9.1', 'chardet==2.2.1', 'click==6.7', 'colorama==0.3.7', 'cryptography==1.7.1', 'docopt==0.6.2', 'enum34==1.1.6', 'fabric==1.13.1', 'idna==2.2', 'iniparse==0.4', 'ipaddress==1.0.18', 'jinja2==2.9.4', 'kaptan==0.5.8', 'kitchen==1.1.1', 'libtmux==0.6.1', 'markupsafe==0.23', 'mysql-connector-python-rf==2.2.1', 'mysql-connector==2.1.4', 'mysql-python==1.2.5', 'paramiko==2.1.1', 'pex==1.1.20', 'pip==9.0.1', 'prettytable==0.7.2', 'psutil==5.0.1', 'pyasn1==0.1.9', 'pycparser==2.17', 'pycurl==7.19.0', 'pygpgme==0.3', 'pyliblzma==0.5.3', 'pyxattr==0.5.1', 'pyyaml==3.12', 'records==0.5.0', 'requests==2.12.5', 'setuptools==30.4.0', 'six==1.10.0', 'sqlalchemy==1.1.5', 'tablib==0.11.3', 'tmuxp==1.2.4']
>>> import mysql.connector
Traceback (most recent call last):
  File "<console>", line 1, in <module>
ImportError: No module named mysql.connector

the mysql-connector has been resolved properly, but I can't use it in pex

Could someone give me a hint?

Thanks,

Xin

**Comment by sixninetynine** [comment]
This looks to be the same issue as https://github.com/pantsbuild/pex/issues/325

Check out where the wheel installs to (when unzipping the pex):

```
...
  inflating: .deps/mysql_connector-2.1.4-cp27-none-macosx_10_12_intel.whl/mysql_connector-2.1.4.data/purelib/mysql/__init__.py
...
```

That purelib path prefix is what's causing the problems, for example if I manually add that path to my `sys.path` I can import the package:

```
osx /var/folders/6z/wjqhl7z52kq3lml9q17qxqg80000gn/T/tmp.yjWTuXdiUM ❯❯❯ pex 'mysql-connector==2.1.4' -o mc.pex -v
pex: Warning, using a UrllibContext which is known to be flaky.
pex: Please build pex with the requests module for more reliable downloads.
  mysql-connector 2.1.4
pex: Building pex: 76.7ms
pex:   Resolving distributions: 33.8ms
Saving PEX file to mc.pex
osx /var/folders/6z/wjqhl7z52kq3lml9q17qxqg80000gn/T/tmp.yjWTuXdiUM ❯❯❯ ./mc.pex
Python 2.7.10 (default, Jul 30 2016, 19:40:32)
[GCC 4.2.1 Compatible Apple LLVM 8.0.0 (clang-800.0.34)] on darwin
Type "help", "copyright", "credits" or "license" for more information.
(InteractiveConsole)
>>> import mysql.connector
Traceback (most recent call last):
  File "<console>", line 1, in <module>
ImportError: No module named mysql.connector
>>> import sys
>>> import os
>>> sys.path.append(os.path.join(os.getcwd(), 'mc.pex', '.deps/mysql_connector-2.1.4-cp27-none-macosx_10_12_intel.whl/mysql_connector-2.1.4.data/purelib')
... )
>>> import mysql.connector
>>> mysql.connector
<module 'mysql.connector' from '/private/var/folders/6z/wjqhl7z52kq3lml9q17qxqg80000gn/T/tmp.yjWTuXdiUM/mc.pex/.deps/mysql_connector-2.1.4-cp27-none-macosx_10_12_intel.whl/mysql_connector-2.1.4.data/purelib/mysql/connector/__init__.py'>
```

So it would appear that pex simply doesn't understand the purelib/platlib stuff at install time (confirmed by @kwlzn's comment in #325), I don't know enough about those dirs or how/why/when they are used to comment further.

**Comment by jsirois** [comment]
Thanks for digging @sixninetynine - I've closed this as duplicate of #325.
